src/main/webapp/app/shared/subject/subject-pair-dialog.component.ts

```diff
@@ -35,9 +37,23 @@ export class SubjectPairDialogComponent implements OnInit, OnDestroy {
                 private oauthClientService: OAuthClientService,
                 private pairInfoService: OAuthClientPairInfoService,
                 private datePipe: DatePipe,
+                private printService: PrintService,
                 @Inject(DOCUMENT) private doc) {
         this.authorities = ['ROLE_USER', 'ROLE_SYS_ADMIN'];
     }
+    async exportHtmlToPDF() {
+        // let pp: PrintService = this.printService;
+        // window.onafterprint = function(){
+        //     console.log("Printing completed...");
+        //     //pp.togglePrintLock().then(_ => {})
+        // }
+
+        this.printService.setPrintLockTo(true);
+        await new Promise(r => setTimeout(r, 0.000000000000000000000000000000000000000000000000000001));
```

Comment:
I think the implementation is fine except for the need for this delay and I cannot figure out how to remove it...

Reply:
It’s needed because otherwise angular cannot do change detection while you’re in a blocking function. Alternatively, you lock the print lock when the modal opens and release when the modal disappears. That way if they use the browser to print, they still get the same view. And we can expect that when users print a modal, they are not interested in the background.

Reply:
Thanks, that fixed it indeed!